Relationships/RelationshipResolver.cs

```diff
@@ -70,49 +71,93 @@ public class RelationshipResolver
+
+
+        public RelationshipResolver(RelationshipResolverOptions options, Registry registry, KSPVersion kspversion)
+        {
+            this.registry = registry;
+            this.kspversion = kspversion;
+            this.options = options;
 
+            installed_modules = new HashSet<Module>(registry.InstalledModules.Select(i_module => i_module.Module));
+            var installed_relationship = new Relationship.Installed();
+            foreach (var module in installed_modules)
+            {
+                reasons.Add(module, installed_relationship);
+            }
+        }
 
-        public RelationshipResolver(ICollection<string> moduleNames, RelationshipResolverOptions options, Registry registry,
+        public RelationshipResolver(IEnumerable<string> module_names, RelationshipResolverOptions options, Registry registry,
             KSPVersion kspversion) :
-                this(moduleNames.Select(name => CkanModule.FromIDandVersion(registry, name, kspversion)).ToList(),
+                this(module_names.Select(name => CkanModule.FromIDandVersion(registry, name, kspversion)).ToList(),
                     options,
                     registry,
                     kspversion)
         {
             // Does nothing, just calles the other overloaded constructor
         }
-
+       
         /// <summary>
         /// Creates a new resolver that will find a way to install all the modules specified.
         /// </summary>
         public RelationshipResolver(ICollection<CkanModule> modules, RelationshipResolverOptions options, Registry registry,
-            KSPVersion kspversion)
+            KSPVersion kspversion):this(options,registry,kspversion)
         {
-            this.registry = registry;
-            this.kspversion = kspversion;
+            AddModulesToInstall(modules);
+        }
+
+        /// <summary>
+        ///     Returns the default options for relationship resolution.
+        /// </summary>
+        
+        // TODO: This should just be able to return a new RelationshipResolverOptions
+        // and the defaults in the class definition should do the right thing.
+        public static RelationshipResolverOptions DefaultOpts()
+        {
+            var opts = new RelationshipResolverOptions
+            {
+                with_recommends = true,
+                with_suggests = false,
+                with_all_suggests = false
+            };
+
+            return opts;
+        }
+
+        /// <summary>
+        /// Add modules to consideration of the relationship resolver. Optional installed parameter defaults
+        /// to installed mods and is intended to be used when this is incorrect, such as when some are to be 
+        /// removed.         
+        /// </summary>
+        /// <param name="modules">Modules to checkattempt to install</param>
+        /// <param name="installed_modules">Currently installed modules to consider</param>
```

Comment:
I might be missing something here, but I can't see `installed_modules` as a parameter in the method declaration...